### DISCUSSION TITLE
Partial Docker Deployment

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#8549] [Q&A]
*Partial Docker Deployment*

I'm trying to work out possible deployment patterns. I'd like to keep the dagster framework components (`dagit`, `dagster-daemon`, `postgres`) independent of the workers and the user code (repos). A workspace of gRPC repos works well for this. I can bring the repos up and down independent of the dagster framework components, e.g. when user code has been changed.

I'm using `docker-compose` to run the `dagit`, `dagster-daemon` and `postgres`, but the worker (repo) runs outside of `docker-compose`, perhaps even on another machine. I'm able to load the jobs in `dagit`, but when I launch a job from `dagit` I get PostgreSQL connection error at the worker (it looks like the `docker-compose` specific hostname from `dagster.yaml` is being used, which is not a valid hostname outside of the `docker-compose` network). Is there a way for me to override the PostgreSQL config for the worker? I'm already setting environment variables:
• `export DAGSTER_POSTGRES_USER="postgres_user"`
• `export DAGSTER_POSTGRES_PASSWORD="postgres_password"`
• `export DAGSTER_POSTGRES_DB="postgres_db"`

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1637140203299400?thread_ts=1637140203.299400&cid=C01U954MEER).*

**Comment by yuhan** [comment]
If changing the environment variable value in the worker an option, you could use `DAGSTER_PG_HOSTNAME` for the 'hostname' field and have it be one thing in dagit and one thing in the worker, as per https://docs.dagster.io/deployment/dagster-instance#postgres-storage

Those env vars are processed when each process that uses them starts up, so you'd be able to use the same dagster.yaml on two different hosts I believed.
